### ISSUE TITLE
BRAT capacity seems low for Western Montana?

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/pyBRAT#172]
Just putting a question here from Claudine who is running BRAT for Montana. Its a good question and may benefit other users....

> On Fri, Sep 21, 2018 at 11:25 AM Tobalske, Claudine <Claudine.Tobalske@mso.umt.edu> wrote:
> Good morning Wally,
> 
> 
> As you know, I’ve been running BRAT for the whole state of Montana, HUC by HUC. Once it was finished, I appended all the HUCs to get the “big picture”, and one thing really stood out – see image below, Dam Building Capacity (oCC_EX) displayed by classes (red = 0, None; orange = 0-1, Rare; yellow = 1-5, Occasional; green = 5-15, Frequent; blue = 15-40, Pervasive). The West hydrologic region (along ID border) clearly stands out with a very high percentage of stream segments classified as Rare Dam Building Capacity.
> 
> 
> I looked into the BRAT outputs, comparing this region with the rest of the state, and also got some stats comparing streams with low oCC_EX values (<=1) with streams with high occ_EX values (>1).  The major difference is values for iHyd_Q2, typical two-year flood; they’re extremely high in this region (mean iHyd_Q2 = 7,600) compared to the rest of the state (mean iHyd_Q2 = 243). The equation, obtained from the Montana StreamStats Report, is 0.131 A0.920 P2.24 (F + 1)-0.845 where A is drainage area, P is mean annual precipitation in inches, and F is % drainage that’s forested.  Compared with Q2 computations for other regions, the main difference seem to stem from the precipitation component of the equation.
> 
![image002](https://user-images.githubusercontent.com/9723905/46026467-a16b4280-c0a8-11e8-8108-acb91a402a88.jpg)
> Hence my question: does the BRAT output for the West hydrologic region represent a real characteristics (a lot of streams with low dam building capacity because of high stream runoff), or is it an artifact?  I tend to lean towards the latter because there are plenty of beaver ponds in that part of Montana, including on streams classified as low dam building capacity.
> 
> Have you run into something similar in Utah, around the Wasatch mountain range, for example?
> 
> Any comments/suggestions would be much appreciated!
> 
> Best,
> Claudine
> 
> Claudine Tobalske, Ph.D. │ GIS Analyst/Ecologist │ Montana Natural Heritage Program (http://mtnhp.org) │ NS 205 The University of Montana, Missoula, MT 59812 │ (406) 243-5196 │ Claudine.tobalske@umontana.edu

**Comment by joewheaton** [comment]
@wally-mac replied:

**Comment by joewheaton** [comment]
Hi Claudine,
Good to see you're getting BRAT run for Montana. I think Wally and your synopsis is likely correct. One easy way to look at this is take a look at the `oVC_EX`  output (i.e. just the vegetation based capacity) in comparison to your `oCC_EX`, which shows the combined capacity (i.e. limiting based on hydrology). Note see [this table](http://brat.riverscapes.xyz/Documentation/Tutorials/WorkingWithBRATOutputs.html#attribute-field-descriptions) for cheat sheet of attributes.  Assuming you've got your vegetation calibration from LANDFIRE dialed, if the `oVC_EX` looks like what you'd expect, then that's helpful context. We've been working a lot with exposing all the intermediates and inputs in the new layer package generator to help you troubleshoot these things.

As for tweaking your regional curves, @bangen is working on some helpful documentation for tips and tricks there. These regional curves can be wildly off, but sometimes its easy to get them messed up just by using the wrong units on these precipitation and slope coefficients some of them add. Remember, the dominant driver is just drainage area, and they add those otehr coefficents in there just to fine-tune. So, take a look  and double check your units.  I would spot check them against what you get on [USGS Streamstats](https://water.usgs.gov/osw/streamstats/) too, to make sure you're calculating them correctly (they're using the same curves). 

Finally, lets use this forum (you can create free account on GItHub) to have the conversation instead of email as other users will likely find it helpful. The problem you're experiencing is very common...  and straightforward to fix. @bangen can you chime in too?

Best Wishes,
Joe

**Comment by ctobalsk** [comment]
Thanks Sara!
for (2), I did use the same publication you posted the link to, to compute typical two-year flood.  For the West hydrologic region (for all regions actually), I used the equation for Q50 [in this case, Q50 = 0.131 A0.920 P2.24 (F + 1)-0.845], not the equation for Q2 [Q2 = 12.2 A0.818 P1.42 (F + 1)-0.896].
The legend for table 1.4 states “QAEP, peak-flow magnitude, in cubic feet per second, for annual exceedance probability (AEP) in percent”; hence a two-year flood would be calculated using a 50% exceedance probability (Q50), whereas a 50 year flood would use a 2% (Q2) exceedance probability.  
So unfortunately, there is no “quick fix” using a different regression equation…

I’ll keep digging!

Claudine

**Comment by ctobalsk** [comment]
FOUND THE PROBLEM!  At least I'm pretty sure I did.  I used proportions, instead of percents, in the regression equation...  By using percents, for example in the Bitterroot HUC4, Q2 drops from over 300,000 CFS to about 15,000 CFS. Much better!!  I'll just need to re-run BRAT python scripts starting at iHyd. Hopefully that'll do it!  Thanks for the support; taking a really close look at the Q2 equation is what revealed the error.
